app/controllers/application_drafts_controller.rb

```diff
@@ -111,14 +122,20 @@ def ensure_max_applications
     end
   end
 
+  def require_student
+    unless application_draft.as_student?
+      redirect_to application_drafts_path, alert: 'You must be listed as a student on your team'
+    end
+  end
+
   def current_team
-    current_student.current_team || coaches_team
+    current_student.current_team || extended_team
   end
 
-  def coaches_team
-    @coaches_team ||= begin
+  def extended_team
```

Comment:
We don't set the team explicitely and instead retrieve it via the `ApplicationDraft` instance (which we find with `params[:id]`). This method (together with `current_team` in line 131) wants to make sure that the team is discarded when the currently logged in user is not part of that team and that we have no start of the association chain when `application_draft` is looked up in line 71.

>  could we consider a refactor?

By all means YES. However, I would very much like to do this after this functional PR if you think this method does what it is supposed to do (see above).